src/libraries/System.Net.Security/tests/FunctionalTests/CertificateValidationRemoteServer.cs

```diff
@@ -15,23 +17,45 @@ namespace System.Net.Security.Tests
 
     public class CertificateValidationRemoteServer
     {
-        [Fact]
         [OuterLoop("Uses external servers")]
-        public async Task CertificateValidationRemoteServer_EndToEnd_Ok()
+        [ConditionalTheory]
+        [InlineData(false)]
+        [InlineData(true)]
+        public async Task CertificateValidationRemoteServer_EndToEnd_Ok(bool useAsync)
         {
-            if (PlatformDetection.IsWindows7)
-            {
-                // https://github.com/dotnet/corefx/issues/42339
-                return;
-            }
-
             using (var client = new TcpClient(AddressFamily.InterNetwork))
             {
-                await client.ConnectAsync(Configuration.Security.TlsServer.IdnHost, Configuration.Security.TlsServer.Port);
+                try
+                {
+                    await client.ConnectAsync(Configuration.Security.TlsServer.IdnHost, Configuration.Security.TlsServer.Port);
+                }
+                catch (Exception ex)
+                {
+                    // if we cannot connect, skip the test instead of failing.
+                    // This test is not trying to test networking.
+                    new SkipTestException($"Unable to connect to '{Configuration.Security.TlsServer.IdnHost}': {ex.Message}");
```

Comment:
I think this should be `throw new SkipTestException()'.  Doing just `new SkipTestException()` won't do anything.

Reply:
good catch, fixed.